daemon/main.go

```diff
@@ -657,6 +635,10 @@ func runDaemon() {
 	d.ciliumHealth = &health.CiliumHealth{}
 	go d.ciliumHealth.Run()
 
+	addressing := d.getNodeAddressing()
+	cancelHealth := health.LaunchAsEndpoint(d, addressing, d.conf.Opts)
+	defer cancelHealth()
```

Comment:
It took me a long time to understand the distinction between `d.ciliumHealth.Run()` and `health.LaunchAsEndpoint()`, mostly because I didn't understand what cilium-health was doing. I could understand what each call did, but not the intent behind them. I'm not sure how to make it more obvious overall, but perhaps comments here explaining what each is and that the `.Run()` spawned one tests endpoint connectivity against the `.LaunchAsEndpoint` spawned one might help a bit. Alternatively, more comments on the intended usecases for each function in it's comment might also work, although that seems like the wrong place to put that.

Reply:
As a brief description here: We launch two instances of cilium-health, one on the host and one which acts as an endpoint. This allows us to test multiple connectivity paths to other nodes in the cluster.

With respect to the code, I can add brief comments here to distinguish the two and I'll think about an appropriate place to put a description like the above.

Reply:
Yes please